src/test/java/com/rocketden/main/service/RoomServiceTests.java

```diff
@@ -210,6 +212,38 @@ public void joinFullRoomFailure() {
     }
 
 
+    @Test
+    public void manyUsersJoiningAnInfinitelySizedRoomSuccess() {
+        /**
+         * Verify join room request fails when the room is already full
+         * Define five users, and add to the room
+         */
+        User firstUser = new User();
+        firstUser.setNickname(NICKNAME);
+        UserDto secondUser = new UserDto();
+        secondUser.setNickname(NICKNAME_2);
+        JoinRoomRequest request = new JoinRoomRequest();
+        request.setUser(secondUser);
+
+        Room room = new Room();
+        room.setRoomId(ROOM_ID);
+        room.setSize((int) (RoomService.MAX_SIZE + 1));
+        room.setHost(firstUser);
+        room.addUser(firstUser);
+
+        for (int i = 0; i < 100; i++) {
+            User temp = new User();
+            temp.setNickname("Rocket" + i);
+            room.addUser(temp);
+        }
+
+        // Mock repository to return room when called
+        Mockito.doReturn(room).when(repository).findRoomByRoomId(eq(ROOM_ID));
+        assertDoesNotThrow(() -> roomService.joinRoom(ROOM_ID, request));
+        verify(repository).findRoomByRoomId(ROOM_ID);
```

Comment:
Perhaps also verify that the number of users is now equal to how many there are in this room - `101` or so. You could replace the `assertDoesNotThrow` with this, or simply keep both in (either works imo).